pkg/configurator/crd_client.go

```diff
@@ -0,0 +1,179 @@
+		for {
+			msg := <-cfgSubChannel
+
+			psubMsg, ok := msg.(events.PubSubMessage)
+			if !ok {
+				log.Error().Msgf("Could not cast pubsub message")
+				continue
+			}
+
+			switch psubMsg.AnnouncementType {
+			case announcements.MeshConfigAdded:
+				log.Debug().Msgf("[%s] OSM MeshConfig added event triggered a global proxy broadcast",
+					psubMsg.AnnouncementType)
+				events.GetPubSubInstance().Publish(events.PubSubMessage{
+					AnnouncementType: announcements.ScheduleProxyBroadcast,
+					OldObj:           nil,
+					NewObj:           nil,
+				})
+
+			case announcements.MeshConfigDeleted:
+				// Ignore deletion. We expect config to be present
+				log.Debug().Msgf("[%s] OSM MeshConfig deleted event triggered a global proxy broadcast",
+					psubMsg.AnnouncementType)
+				events.GetPubSubInstance().Publish(events.PubSubMessage{
+					AnnouncementType: announcements.ScheduleProxyBroadcast,
+					OldObj:           nil,
+					NewObj:           nil,
+				})
+
+			case announcements.MeshConfigUpdated:
+				// Get config map
+				prevMeshConfigObj, okPrevCast := psubMsg.OldObj.(*v1alpha1.MeshConfig)
+				newMeshConfigObj, okNewCast := psubMsg.NewObj.(*v1alpha1.MeshConfig)
+				if !okPrevCast || !okNewCast {
+					log.Error().Msgf("[%s] Error casting old/new MeshConfigs objects (%v %v)",
+						psubMsg.AnnouncementType, okPrevCast, okNewCast)
+					continue
+				}
+
+				// Parse old and new configs
+				prevMeshConfig := parseOSMMeshConfig(prevMeshConfigObj)
+				newMeshConfig := parseOSMMeshConfig(newMeshConfigObj)
+
+				// Determine if we should issue new global config update to all envoys
+				triggerGlobalBroadcast := false
+
+				triggerGlobalBroadcast = triggerGlobalBroadcast || (prevMeshConfig.Egress != newMeshConfig.Egress)
+				triggerGlobalBroadcast = triggerGlobalBroadcast || (prevMeshConfig.PermissiveTrafficPolicyMode != newMeshConfig.PermissiveTrafficPolicyMode)
+				triggerGlobalBroadcast = triggerGlobalBroadcast || (prevMeshConfig.UseHTTPSIngress != newMeshConfig.UseHTTPSIngress)
+				triggerGlobalBroadcast = triggerGlobalBroadcast || (prevMeshConfig.TracingEnable != newMeshConfig.TracingEnable)
+				triggerGlobalBroadcast = triggerGlobalBroadcast || (prevMeshConfig.TracingAddress != newMeshConfig.TracingAddress)
+				triggerGlobalBroadcast = triggerGlobalBroadcast || (prevMeshConfig.TracingEndpoint != newMeshConfig.TracingEndpoint)
+				triggerGlobalBroadcast = triggerGlobalBroadcast || (prevMeshConfig.TracingPort != newMeshConfig.TracingPort)
+
+				if triggerGlobalBroadcast {
+					log.Debug().Msgf("[%s] OSM MeshConfig update triggered global proxy broadcast",
+						psubMsg.AnnouncementType)
+					events.GetPubSubInstance().Publish(events.PubSubMessage{
+						AnnouncementType: announcements.ScheduleProxyBroadcast,
+						OldObj:           nil,
+						NewObj:           nil,
+					})
+				} else {
+					log.Trace().Msgf("[%s] OSM MeshConfig update, NOT triggering global proxy broadcast",
+						psubMsg.AnnouncementType)
```

Comment:
This seems like a copy-paste from the existing configmap based implementation, and there have been some recent bugs and fixes in this area. 

Would it be possible to consolidate this duplicate code?

/cc @eduser25

Reply:
I referred to my message with @draychev last week that, in v0.9 we will directly switch to CRD, rather than supporting both configmap and CRD. Thus if soon we will remove configmap client completely, I think it's fine not to extract it out now.

Reply:
That's okay. @eduser25 could you check if the recent fixes in configurator are present here?

Reply:
lgtm so far